staging/src/k8s.io/apiserver/pkg/server/genericapiserver.go

```diff
@@ -242,12 +242,8 @@ func (s *GenericAPIServer) MinRequestTimeout() time.Duration {
 	return s.minRequestTimeout
 }
 
-type preparedGenericAPIServer struct {
-	*GenericAPIServer
-}
-
 // PrepareRun does post API installation setup steps.
-func (s *GenericAPIServer) PrepareRun() preparedGenericAPIServer {
+func (s *GenericAPIServer) PrepareRun() *GenericAPIServer {
```

Comment:
this change is not ok. The hidden type `preparedGenericAPIServer` is intentional to enforce that the Prepare method is called.

Reply:
The worst case is forget call PrepareRun, which is a definitely coding bug. And can be checked by healthz check.

Reply:
Many developers read these code have a doubt, why coding like this ?  Remove the wrapper just make it more readable.

Reply:
the point is that this allows static checking. The compiler complains. No reason to remove this. It costs nothing and avoids mistakes early.

Reply:
Add a comment if it is not clear.

Reply:
agree